### PR TITLE
0.4 akeloth postgresql

### PR DESCRIPTION
**Pull request by abhilash-keloth**
fixed workflow

### CODE REVIEW

File: examples/postgresql_server/100-simple-postgresql/configuration.tfvars [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+postgresql_servers = {
+  sales-re1 = {
+    name                          = "sales-re1"
+    region                        = "region1"
+    resource_group_key            = "postgresql_region1"
+    version                       = "9.6"
+    sku_name                      = "GP_Gen5_2"
+    administrator_login           = "postgresqlsalesadmin"
+    administrator_login_password  = "Testpass@9"
```
Comment (LaurentLesle) [inline examples/postgresql_server/100-simple-postgresql/configuration.tfvars:19]: Comment the Admin password to get the password auto-generated
The comment will show the option to pass the password if needed

File: examples/postgresql_server/100-simple-postgresql/configuration.tfvars [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+keyvaults = {
+  postgresql-re1 = {
+    name               = "postgresqlre1"
+    resource_group_key = "security_region1"
+    sku_name           = "standard"
+  }
+}
+
+keyvault_access_policies = {
```
Comment (LaurentLesle) [inline examples/postgresql_server/100-simple-postgresql/configuration.tfvars:123]: Move the two policies as creations policies in the keyvault.

See examples https://github.com/aztfmod/terraform-azurerm-caf/blob/c32d1f5964a3322067414446ef31c347c8c148b6/examples/mssql_server/200-mssql.tfvars#L62

File: examples/postgresql_server/100-simple-postgresql/configuration.tfvars [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    logged_in_user = {
+      secret_permissions = ["Set", "Get", "List", "Delete", "Purge"]
+    }
+    logged_in_aad_app = {
+      secret_permissions = ["Set", "Get", "List", "Delete", "Purge"]
+    }
+  }
+}
+
+azuread_groups = {
```
Comment (LaurentLesle) [inline examples/postgresql_server/100-simple-postgresql/configuration.tfvars:135]: Can you attach the azure ad group the the server?

Comment (LaurentLesle) [reply examples/postgresql_server/100-simple-postgresql/configuration.tfvars:135]: or remove it if not used

File: examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+postgresql_servers = {
+  sales-re1 = {
+    name                          = "sales-re1"
+    region                        = "region1"
+    resource_group_key            = "postgresql_region1"
+    version                       = "9.6"
+    sku_name                      = "GP_Gen5_2"
+    administrator_login           = "postgresqlsalesadmin"
+    administrator_login_password  = "Testpass@9"
```
Comment (LaurentLesle) [inline examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars:19]: Same comment as before

File: examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+keyvaults = {
+  postgresql-re1 = {
+    name               = "postgresqlre1"
+    resource_group_key = "security_region1"
+    sku_name           = "standard"
+  }
+}
+
+keyvault_access_policies = {
```
Comment (LaurentLesle) [inline examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars:124]: same

File: examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    logged_in_user = {
+      secret_permissions = ["Set", "Get", "List", "Delete", "Purge"]
+    }
+    logged_in_aad_app = {
+      secret_permissions = ["Set", "Get", "List", "Delete", "Purge"]
+    }
+  }
+}
+
+azuread_groups = {
```
Comment (LaurentLesle) [inline examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars:136]: same

File: modules/databases/postgresql_server/server.tf [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+resource "azurerm_key_vault_secret" "sql_admin" {
+  count = try(var.settings.administrator_login_password, null) == null ? 1 : 0
+
+  name         = format("%s-username", azurecaf_name.postgresql.result)
+  value        = var.settings.administrator_login
+  key_vault_id = var.keyvault_id
+}
+
+resource "azurerm_postgresql_active_directory_administrator" "aad_admin" {
```
Comment (LaurentLesle) [inline modules/databases/postgresql_server/server.tf:79]: make sure you have an example to test this feature

**Review by LaurentLesle — CHANGES_REQUESTED** [review]
Thanks @abhilash-keloth for the PR. Great stuff in it. I have added my comments. Can you address them?

File: examples/postgresql_server/100-simple-postgresql/configuration.tfvars (hunk 1 above)
Comment (abhilash-keloth) [reply examples/postgresql_server/100-simple-postgresql/configuration.tfvars]: done

File: examples/postgresql_server/100-simple-postgresql/configuration.tfvars (hunk 2 above)
Comment (abhilash-keloth) [reply examples/postgresql_server/100-simple-postgresql/configuration.tfvars]: done

File: examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars (hunk 5 above)
Comment (abhilash-keloth) [reply examples/postgresql_server/101-active-directory-admin-postgresql/configuration.tfvars]: done

**Pull request closed without merging**